source/faq.rst

```diff
@@ -0,0 +1,178 @@
+
+Contextualisation is particularly useful for:
+
+* Configuration not known until instantiation (e.g. data location).
+
+* Private Information (e.g. host certs)
+
+* Software that changes frequently or under development.
+
+The contextualisation requires passing some data to the VMs on instantiation
+(the context) and handling that context in the VM.
+
+How can I inject my public SSH key into the machine?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The best way to login into the virtual server is to use SSH keys. If you don't
+have one, you need to generate it with the ``ssh-keygen`` command:
+
+.. code-block:: console
+
+   ssh-keygen -f fedcloud
+
+This will generate two files:
+
+* ``fedcloud``, the private key. This file should never be shared
+
+* ``fedcloud.pub``, the public key. That will be sent to your VM.
+
+To inject the public SSH key into the VM you can use the `key-name` option
+when creating the VM in OpenStack. Check `keypair management <https://docs.openstack.org/python-openstackclient/pike/cli/command-objects/keypair.html>`_
+option in OpenStack documentation. This key will be available for the default
+configured user of the VM (e.g. ``ubuntu`` for Ubuntu, ``centos`` for CentOS).
+
+You can also create users with keys with a contextualisation file:
+
+.. code-block:: yaml
+
+   #cloud-config
+   users:
+     - name: cloudadm
+       sudo: ALL=(ALL) NOPASSWD:ALL
+       lock-passwd: true
+       ssh-import-id: cloudadm
+       ssh-authorized-keys:
+         - <paste here the contents of your SSH key pub file>
+
+.. warning::
+
+   YAML format requires that the spaces at the beginning of each line is
+   respected in order to be correctly parsed by ``cloud-init``.
+
+
+How can I use a contextualisation file?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have a contextualisation file, you can use it with the ``--user-data``
+option to ``server create`` in OpenStack.
+
+.. code-block:: shell
+
+   $ openstack server create --flavor <your-flavor> --image <your image> \
+               --user-data <your contextualisation file> \
+               <server name>
+
+.. note::
+
+   We recommend using ``cloud-init`` for contextualisation. EGI images in AppDB
+   do support ``cloud-init``. Check the `documentation <http://cloudinit.readthedocs.org/ cloud-init>`_
+   for more information.
+
+How can I pass secrets to my VMs?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+EGI Cloud endpoints use https so information passed to contextualise the VMs
+can be assumed to be safe and only readable within your VM. However, take into
+account that anyone with access to the VM may be able to access also the
+contextualisation information.
+
+The disk on my VM is full, how can I get more space?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are several ways to increase the disk space available at the VM. The
+fastest and easiest one is to use block storage, creating a new storage disk
```

Comment:
A link to related documentation would be nice

Reply:
Fixed